modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/ArtifactExtensionManagerFactory.java

```diff
@@ -33,12 +29,16 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.BiFunction;
 
 /**
  * Creates {@link ExtensionManager} for mule artifacts that own a {@link MuleContext}
  */
 public class ArtifactExtensionManagerFactory implements ExtensionManagerFactory {
 
+  public static final BiFunction<PluginClassLoaderSupplier, ExtensionModelLoaderRepository, ExtensionModelDiscoverer> EXT_MODEL_DISCOVERER =
```

Comment:
Why is this defined here and used in ExtensionFacctories that doesn't extend this class. Can we refactor this hierarchy so that this is not imported in PolicyExtensionManagerFactory. I think we should redefine the hierarchy of PolicyExtensionManagerFactory, ArtifactExtensionManagerFactory so that this static imports are not needed.